src/core/server/saved_objects/export/get_sorted_objects_for_export.ts

```diff
@@ -84,6 +84,9 @@ async function fetchObjectsToExport({
   savedObjectsClient: SavedObjectsClientContract;
   namespace?: string;
 }) {
+  if ((types?.length ?? 0) > 0 && (objects?.length ?? 0) > 0) {
+    throw Boom.badRequest(`Can't specify both "types" and "objects" properties when exporting`);
+  }
```

Comment:
The route validation was using an `nand(types, objects)`. As we don't have the equivalent in config-schema (probably for the best), I moved this check in the actual service method. This probably is it's place anyway.

I also had to fix some calls that were actually sending both (with types being used as they are exclude in the business logic)